playground/src/castAction.tsx

```diff
@@ -36,18 +39,82 @@ export const app = new Frog()
         </div>
       ),
       intents: [
-        <Button.AddCastAction action="/action" name="Log This!" icon="log">
-          Add
+        <Button.AddCastAction action="/action-message">
+          Message
+        </Button.AddCastAction>,
+        <Button.AddCastAction action="/action-frame">
+          Frame
         </Button.AddCastAction>,
       ],
     }),
   )
-  .castAction('/action', async (c) => {
-    console.log(
-      `Cast Action to ${JSON.stringify(c.actionData.castId)} from ${
-        c.actionData.fid
-      }`,
-    )
-    if (Math.random() > 0.5) return c.error({ message: 'Action failed :(' })
-    return c.res({ message: 'Action Succeeded' })
+  .castAction(
+    '/action-message',
+    async (c) => {
+      console.log(
+        `Cast Action to ${JSON.stringify(c.actionData.castId)} from ${
+          c.actionData.fid
+        }`,
+      )
+      if (Math.random() > 0.5) return c.error({ message: 'Action failed :(' })
+      return c.res({ type: 'message', message: 'Action Succeeded' })
```

Comment:
I wonder if separating frame responses to `c.frame(...)` would make more sense to not have a breaking change here.

Reply:
hm, as I think about it it makes more sense, as we have similar specific response handler `error`, and `res` being preserved as "traditional" response for such – just as cast actions also responded with messages traditionally.